packages/shared/src/introspectMachine.ts

```diff
@@ -188,6 +188,10 @@ export const introspectMachine = (machine: XState.StateNode) => {
     });
 
     node.transitions?.forEach((transition) => {
+      node.onExit.forEach((action) => {
+        actions.addEventToItem(action.type, transition.eventType, node.path);
+      });
```

Comment:
Hm, I'm not entirely sure but this one seems to be a little bit simplistic. I didn't test it but if I read correctly how this works then it will not work properly for cases like this:
```js
createMachine({
  states: {
    a: {
      exit: 'myAction', // this should receive FOO | 'xstate.stop'
      states: {
        b: {
          on: {
            FOO: '#someIdElsewhere'
          }
        }
      }
    }
  }
})
```

Reply:
The algorithm should account for targets within the subtree being outside of the "source" state that has the `exit` handler defined on it. We should also recheck if this correctly works with internal/external transitions.

Reply:
👍  Good point

Reply:
@Andarist Added a test case to cover this

Reply:
An additional test case that also probably fails is this:
```js
createMachine({
  states: {
    a: {
      on: {
	    FOO: '#someIdElsewhere'
	  },
      states: {
        b: {
          exit: 'myAction', // this should receive FOO | 'xstate.stop'
        }
      }
    }
  }
})
```

Reply:
Nice, good catch. Fixed